Detector/DetFCCeeIDEA/src/Mask_o1_v01_geo.cpp

```diff
@@ -0,0 +1,229 @@
+    case ODH::kPunchedDnstream: {
+      // a volume on the upstream or downstream branch with two inner holes
+      // (implemented as a cone from which another tube is punched out)
+
+      const double rCenterPunch = (crossType == ODH::kPunchedUpstream)
+          ? (rInnerStart)
+          : (rInnerEnd);  // just alias names denoting what is meant here
+      const double rOffsetPunch = (crossType == ODH::kPunchedDnstream)
+          ? (rInnerStart)
+          : (rInnerEnd);  // (the database entries are "abused" in this case)
+
+      // relative transformations for the composition of the SubtractionVolumes
+      Transform3D punchTransformer(RotationY(-2 * rotateAngle), Position(zPosition * tan(-2 * rotateAngle), 0, 0));
+      Transform3D punchTransmirror(RotationY(+2 * rotateAngle), Position(zPosition * tan(+2 * rotateAngle), 0, 0));
+
+      // absolute transformations for the final placement in the world
+      Transform3D placementTransformer(RotationY(rotateAngle), RotateY(Position(0, 0, zPosition), rotateAngle));
+      Transform3D placementTransmirror(RotationY(mirrorAngle), RotateY(Position(0, 0, zPosition), mirrorAngle));
+
+      // the main solid and the piece (only a tube, for the moment) which will be punched out
+      ConeSegment wholeSolid(zHalf, rCenterPunch, rOuterStart, rCenterPunch, rOuterEnd, phi1, phi2);
+      Tube punchSolid(0, rOffsetPunch, 5 * zHalf, phi1, phi2);  // a bit longer
+
+      // the punched subtraction solids can be asymmetric and therefore have to be created twice:
+      // one time in the "right" way, another time in the "reverse" way, because the "mirroring"
+      // rotation around the y-axis will not only exchange +z and -z, but also +x and -x
+      SubtractionSolid finalSolid0(wholeSolid, punchSolid, punchTransformer);
+      SubtractionSolid finalSolid1(wholeSolid, punchSolid, punchTransmirror);
+
+      // tube consists of vacuum (will later have two different daughters)
+      Volume tubeLog0(volName + "_0", finalSolid0, material);
+      Volume tubeLog1(volName + "_1", finalSolid1, material);
+      tubeLog0.setVisAttributes(theDetector, xmlMask.visStr());
+      tubeLog1.setVisAttributes(theDetector, xmlMask.visStr());
+
+      // placement of the tube in the world, both at +z and -z
+      envelope.placeVolume(tubeLog0, placementTransformer);
+      envelope.placeVolume(tubeLog1, placementTransmirror);
+
+      break;
+    }
+    default: { throw std::runtime_error(" Mask_o1_v01_geo.cpp : fatal failure !! ??  "); }
+
+    }  // end switch
+  }    // for all xmlSections
+
+  //--------------------------------------
+  Volume mother = theDetector.pickMotherVolume(tube);
+  PlacedVolume pv(mother.placeVolume(envelope));
+  pv.addPhysVolID("system", xmlMask.id());  //.addPhysVolID("side", 0 ) ;
+
+  tube.setVisAttributes(theDetector, xmlMask.visStr(), envelope);
+
+  tube.setPlacement(pv);
+
+  return tube;
+}
+DECLARE_DEPRECATED_DETELEMENT(Mask_o1_v01, create_element)
```

Comment:
I took this driver from the CLIC community (lcgeo). Depricated means that it is not maintained anymore. The plan is to move this driver to DD4hep, but since it is not yet there, I would keep this driver here.

Reply:
The driver has been in DD4hep for some time now https://github.com/AIDASoft/DD4hep/blob/master/DDDetectors/src/Mask_o1_v01_geo.cpp

Reply:
The driver for mask in DD4hep is different from the one I used. For example, the phi angle in `DD4hep_Mask_o1_v01` is hard-coded to 360 degrees whereas we need only a partial coverage. Anna is working to submit the changes to DD4hep to have a generic one which we can use as well. So, for the moment, I keep this file here.